posthog/tasks/tasks.py

```diff
@@ -40,11 +41,14 @@ def redis_heartbeat() -> None:
     autoretry_for=(
         # Important: Only retry for things that might be okay on the next try
         CHQueryErrorTooManySimultaneousQueries,
+        CeleryConcurrencyLimitExceeded,
     ),
-    retry_backoff=1,
-    retry_backoff_max=2,
-    max_retries=3,
+    retry_backoff=True,
+    retry_backoff_max=10,
+    max_retries=20,
 )
+@limit_concurrency(70)  # Do not go above what CH can handle
+@limit_concurrency(5, key=lambda *args, **kwargs: args[0])  # Do not run too many queries at once for the same team
```

Comment:
This is probably a bit low, but 10 is probably good. Is there a way of adding some logging to see how often we hit these numbers?

Reply:
Good point. Wanted to add some metrics that we can add to the async dashboard.